subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestExecutionListener.java

```diff
@@ -65,18 +65,20 @@
  * Most importantly, it will map assertion and platform failures to Gradle's {@link TestFailure} class, which we can send through the TAPI.
  */
 @NonNullApi
-public class JUnitPlatformTestExecutionListener implements TestExecutionListener, RootAssertionToFailureMapper {
+public class JUnitPlatformTestExecutionListener implements TestExecutionListener {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(JUnitPlatformTestExecutionListener.class);
 
-    private final static List<FailureMapper> MAPPERS = Arrays.asList(
+    private final static List<TestFailureMapper> MAPPERS = Arrays.asList(
         new OpenTestAssertionFailedMapper(),
         new OpenTestMultipleFailuresErrorMapper(),
-        new JUnitComparisonFailureMapper(),
+        new JUnitComparisonTestFailureMapper(),
         new AssertjMultipleAssertionsErrorMapper(),
         new AssertErrorMapper()
     );
 
+    private static final DefaultThrowableToTestFailureMapper FAILURE_MAPPER = new DefaultThrowableToTestFailureMapper(MAPPERS);
```

Comment:
The `MAPPERS` field is only used here and can be inlined.

Reply:
I might keep it separate so it's more obvious that that is _the_ list of mappers.